src/AWS.Deploy.Recipes/CdkTemplates/ConsoleAppECSFargateService/Generated/Configurations/AutoScalingConfiguration.cs

```diff
@@ -0,0 +1,46 @@
+// Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+using System;
+using System.Collections.Generic;
+using System.Text;
+
+// This is a generated file from the original deployment recipe. It contains properties for
+// all of the settings defined in the recipe file. It is recommended to not modify this file in order
+// to allow easy updates to the file when the original recipe that this project was created from has updates.
+// This class is marked as a partial class. If you add new settings to the recipe file, those settings should be
+// added to partial versions of this class outside of the Generated folder for example in the Configuration folder.
+
+namespace ConsoleAppEcsFargateService.Configurations
+{
+    public partial class AutoScalingConfiguration
```

Comment:
I observed a weird behavior working with @96malhar, even though auto scaling was not configured, still metadata in the CF template had auto scaling configs due to default values.

Can Cpu* and Memory* be nullable? or something like in L36?

Reply:
Since we settings have default values in the recipe file the values will always have a value here as well and be serialized. We would need to develop a system that allows us to control serialization based on the `DependsOn` settings to make it so the non-used values don't get put in the CF template.

@96malhar issue turned out not to have anything to do with the the non-used values being put in the template and it was a bug in how we were serializing enums.

Reply:
There are multiple ways to solve this, 
- Make serialization intelligent (as you suggested)
- Don't assign default values to the actual configuration rather use some other constant if the config value is not set


Something like
```
public string? Value { get; set; }
public const string ValueDefault = "MyDefaultValue";
```

and use conditional logic to get the final value. This should be easy to do and avoid unnecessary data in CF template.

Looking at the [CDK API](https://docs.aws.amazon.com/cdk/api/latest/docs/@aws-cdk_aws-ecs.RequestCountScalingProps.html#scaleincooldown), they have default values already same as CDK template i.e. 300. Can [ScaleInCooldown ](https://docs.aws.amazon.com/cdk/api/latest/docs/@aws-cdk_aws-ecs.RequestCountScalingProps.html#scaleincooldown)and [ScaleOutCooldown ](https://docs.aws.amazon.com/cdk/api/latest/docs/@aws-cdk_aws-ecs.RequestCountScalingProps.html#scaleincooldown)defaults be removed? Is there a reason that deploy tool has to keep a different default than CDK?

Reply:
The default values are coming from the recipe json file. Whether I put defaults here doesn't matter the process of creating the appsettings.json file from the orchestrator to the CDK project will put the default values in it. I just put the default values in the CDK code as well to avoid checking that will be null. We no they won't be null because the recipe always passes in a value.

Reply:
If default values are coming from recipe files, adding default values in C# code here can hide issues. You can use force unwrapping using `!` for optional fields/properties if it is 100% guaranteed that they will not be null.

Caution: on a null value, the app will crash on a force unwrap.